tests/rptest/utils/si_utils.py

```diff
@@ -288,15 +290,22 @@ def __init__(self, brokers, name, logger, timeout_sec: float = 60.0):
         self.reconnect()
 
     def reconnect(self):
-        self.producer = confluent_kafka.Producer({
-            'bootstrap.servers':
-            self.brokers,
-            'transactional.id':
-            self.name,
-            'transaction.timeout.ms':
-            5000,
-        })
-        self.producer.init_transactions(self.timeout_sec)
+        def init():
+            try:
+                self.producer = ck.Producer({
+                    'bootstrap.servers': self.brokers,
+                    'transactional.id': self.name,
+                    'transaction.timeout.ms': 5000,
+                })
+                self.producer.init_transactions(self.timeout_sec)
+                return True
+            except ck.cimpl.KafkaException as e:
```

Comment:
docs say this, https://docs.confluent.io/platform/current/clients/confluent-kafka-python/html/index.html#producer

```
KafkaError: Use exc.args[0].retriable() to check if the operation may be retried, else treat the error as a fatal error.
```

I think we can generalize this with ^^ 

ex: https://github.com/confluentinc/confluent-kafka-python/blob/e671bccb8a4f98302748ccf60d5d579f68c6613d/tests/test_Producer.py#L202

Reply:
I prefer this way:
  - even fatal errors are retriable during initialization because we haven't send any data
  - also I want to use the fix to document which errors Redpanda throws (if a client tolerates A,B and C while Redpanda throws only C it's valuable information to know)